src/raw/mod.rs

```diff
@@ -1291,48 +1291,52 @@ impl<T, A: Allocator> RawTable<T, A> {
         &mut self,
         hashes: [u64; N],
         eq: impl FnMut(usize, &T) -> bool,
-    ) -> Option<[&'_ mut T; N]> {
+    ) -> [Option<&'_ mut T>; N] {
         unsafe {
-            let ptrs = self.get_many_mut_pointers(hashes, eq)?;
+            let ptrs = self.get_many_mut_pointers(hashes, eq);
 
             for (i, &cur) in ptrs.iter().enumerate() {
-                if ptrs[..i].iter().any(|&prev| ptr::eq::<T>(prev, cur)) {
-                    return None;
+                if let Some(cur) = cur {
+                    if ptrs[..i]
+                        .iter()
+                        .flatten()
+                        .any(|&prev| ptr::eq::<T>(prev, cur))
+                    {
+                        panic!("duplicate keys found");
+                    }
                 }
```

Comment:
Actually, I think you could further change to not destructure `&cur` in the pattern before, but I'm not going to bother writing a larger "suggested change" here.

Reply:
I applied your suggested change. Regarding your second point, I'm not sure I'm following, did you mean `(i, Some(cur))`?

Reply:
I meant changing `for (i, &cur) in` to `(i, cur)`, because `is_some` and `contains` both need that reference anyway. It shouldn't matter much though, LLVM can surely optimize it either way.

Reply:
Done anyway.